### PR TITLE
patrons: add new patron fields

### PR DESCRIPTION
**Pull request by jma**
* Adds notes to user resource.
* Adds expiration date in the patron informations.
* Adds libraries affilation in the patron informations.
* Removes duplicates patron custom validation.
* Groups all patron informations in a nested patron structure.
* Cleans patron tests.
* Displays expiration date and notes in the patron profile.
* Adds web API to return the number of patrons given a username.

Signed-off-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Dependencies:

- rero/rero-ils-ui#385

## Why are you opening this PR?

- https://tree.taiga.io/project/rero21-reroils/us/1590?milestone=273464

## How to test?

- Login with simonetta with the username instead of the email.
- Check in the profile the note below the tabs and the presence of the expiration date in the personal information.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by rerowep — COMMENTED** [review]
`patron.get('patron', {})` is used a lot, it would be nice to have a function or property `patron.patron` for this.

File: tests/api/ill_requests/test_ill_requests_rest.py [hunk 1]
```diff
@@ -71,8 +71,6 @@ def test_ill_requests_get(client, ill_request_martigny):
     data = get_json(res)
 
     metadata = data['hits']['hits'][0]['metadata']
-    from pprint import pprint
-    pprint(metadata)
```
Comment (zannkukai) [inline tests/api/ill_requests/test_ill_requests_rest.py:75]: woops 🙈

**Review by zannkukai — APPROVED** [review]

File: tests/api/test_availability.py [hunk 2]
```diff
@@ -215,6 +221,7 @@ def test_item_holding_document_availability(
             transaction_user_pid=librarian_martigny_no_email.pid
         )
     )
+    print(res)
```
Comment (AoNoOokami) [inline tests/api/test_availability.py:224]: I think we can remove this

**Review by BadrAly — APPROVED** [review]

**Comment by ManaDeweerdt** [comment]
I stay connected even if I log out from the professional interface.
It redirects me to the public interface with an error message "page not found' but I can still navigate.

To reproduce:  login as Virgile. Go to the professional interface and log out.
Note: The url of the logout is different.

**Comment by ManaDeweerdt** [comment]
Requests section from the professional interface : the column 'Item, Location, Requested by, Pick-up Location, Request date, are empty. Only the title of the document is displayed.

**Comment by ManaDeweerdt** [comment]
On a barcode detail view, if there is a request, I can see the patron name but the user barcode is empty.
Furthermore, If I click on the patron name, I'm redirected to a blank page instead of the checkout interface.
image

**Review by pronguen — COMMENTED** [review]
- [ ] A white line is added on every public page and not wanted
image
- [ ] When I log out from pro interface, I stay connected.
- [ ] Is it possible to add a Checkin/checkout note (not public, display at checkin and checkout)?
- [ ] At login, placeholder should be "E-mail or username"
- [ ] After modifying the profile (changing username in personal data) as a patron, I get a message "Internal server error". The username has been modified because I cannot use the previous one anymore. But the new username is not displayed in the *patron profile* (public and pro views). Is there a synchro problem? Or is it out of the scope of this US?
- [ ] It would be more consistent to display the notes below the blocking note (see screenshot). This concerns the pro views (patron detailed view + patron account) and the public view as a patron.
image
- [ ] E-mail should be repetitive, UCL requirement I think (can be addressed in the next US)
- [ ] Should we rename "Profile" in "RERO ID" now?
- [ ] The checkin form performs the checkins but does not display anything (see screenshot).
image
- [ ] It would make it clearer to put also a box "Librarian information" (with only the library for now)
- [ ] Label "Patron Informations" should be "Patron information" (without plural, without uppercase "i")

**Comment by jma** [comment]
> Requests section from the professional interface : the column 'Item, Location, Requested by, Pick-up Location, Request date, are empty. Only the title of the document is displayed.

Hello can you give me link? Thanks.

**Comment by pronguen** [comment]
> 
> 
> > Requests section from the professional interface : the column 'Item, Location, Requested by, Pick-up Location, Request date, are empty. Only the title of the document is displayed.
> 
> Hello can you give me link? Thanks.

https://ilsdev.test.rero.ch/professional/circulation/requests
image

**Comment by jma** [comment]
> * [x]  A white line is added on every public page and not wanted
>   image
> * [x]  When I log out from pro interface, I stay connected.
> * [x]  At login, placeholder should be "E-mail or username"
> * [x]  After modifying the profile (changing username in personal data) as a patron, I get a message "Internal server error". The username has been modified because I cannot use the previous one anymore. But the new username is not displayed in the _patron profile_ (public and pro views). Is there a synchro problem? Or is it out of the scope of this US?
> * [x]  It would be more consistent to display the notes below the blocking note (see screenshot). This concerns the pro views (patron detailed view + patron account) and the public view as a patron.
>   image
> * [x]  Is it possible to add a Checkin/checkout note (not public, display at each checkin, and once on checkout user account)?
> * [ ]  E-mail should be repetitive, UCL requirement I think (can be addressed in the next US)
> * [x]  Should we rename "Profile" in "RERO ID" now?
> * [x]  The checkin form performs the checkins but does not display anything (see screenshot).
>   image
> * [x]  When I enter a patron barcode in the checkin form, it should open this patron account. It spins without doing it.
> * [x]  Request page is broken.
>   image
> * [x]  It would make it clearer to put also a box "Librarian information" (with only the library for now)
> * [x]  Label "Patron Informations" should be "Patron information" (without plural, without uppercase "i")



> I stay connected even if I log out from the professional interface.
> It redirects me to the public interface with an error message "page not found' but I can still navigate.
> 
> To reproduce: login as Virgile. Go to the professional interface and log out.
> Note: The url of the logout is different.

Done

**Pull request merged**